### PR TITLE
[Text QA] Paragraph styles

### PR DESCRIPTION
**Pull request by eessex**
Fixes https://artsyproduct.atlassian.net/browse/GROW-974

- Passes `allowedStyles` prop correctly to Paragraph
- Converts `EditImage` to typescript
- Converts `SectionFooter` to typescript
- Fixes x'd test in `EditImage`

### CODE REVIEW

File: src/client/apps/edit/components/content/sections/images/components/edit_image.tsx [hunk 1]
```diff
@@ -1,20 +1,20 @@
...
-import { Paragraph } from "client/components/draft/paragraph/paragraph"
-import { RemoveButton } from "client/components/remove_button"
 import { clone, without } from "lodash"
 import React, { Component } from "react"
 import { connect } from "react-redux"
 import styled from "styled-components"
+import {
+  onChangeHero,
+  onChangeSection,
+} from "../../../../../../../actions/edit/sectionActions"
```
Comment (xtina-starr) [inline src/client/apps/edit/components/content/sections/images/components/edit_image.tsx]: Why are the paths needing to change here? Are they in different directories now?

File: src/client/apps/edit/components/content/sections/footer/index.tsx [hunk 2]
```diff
@@ -51,15 +51,15 @@ export default connect(
   mapDispatchToProps
 )(SectionFooter)
 
-const FooterContainer = styled.div<{ layout?: string }>`
+const FooterContainer = styled.div.attrs<{ layout?: string }>({})`
```
Comment (xtina-starr) [inline src/client/apps/edit/components/content/sections/footer/index.tsx]: Noticed you switched from the newer styled component syntax here to the newer one. Any particular reason?

File: src/client/apps/edit/components/content/sections/images/components/edit_image.tsx (hunk 1 above)
Comment (xtina-starr) [reply src/client/apps/edit/components/content/sections/images/components/edit_image.tsx]: I see the location didn't change but the file type did.

**Comment by xtina-starr** [comment]
This looks good for the most part. I did have a few question.

File: src/client/apps/edit/components/content/sections/images/components/edit_image.tsx (hunk 1 above)
Comment (eessex) [reply src/client/apps/edit/components/content/sections/images/components/edit_image.tsx]: Yeah tsx files are not resolving absolute paths correctly during test runs, going to make a ticket to resolve this issue separately

File: src/client/apps/edit/components/content/sections/images/components/edit_image.tsx [hunk 3]
```diff
@@ -1,20 +1,20 @@
...
+import { Paragraph } from "../../../../../../../components/draft/paragraph/paragraph"
+import { RemoveButton } from "../../../../../../../components/remove_button"
 
 interface ArticleImage {
-  url?: string
-  caption?: string
+  url?: any
+  caption?: any
```
[the remark this answers is not in the document]
Comment (eessex) [reply src/client/apps/edit/components/content/sections/images/components/edit_image.tsx]: Oh good catch- should be string

File: src/client/apps/edit/components/content/sections/footer/index.tsx (hunk 2 above)
Comment (eessex) [reply src/client/apps/edit/components/content/sections/footer/index.tsx]: The new style is failing in tests where types all appear as undefined, I think we need to update styled components

File: src/client/apps/edit/components/content/sections/images/components/edit_image.tsx (hunk 1 above)
Comment (xtina-starr) [reply src/client/apps/edit/components/content/sections/images/components/edit_image.tsx]: 👍

**Pull request merged**